java-vtl-script/src/main/java/no/ssb/vtl/script/operations/join/InnerJoinOperation.java

```diff
@@ -56,35 +62,33 @@ public InnerJoinOperation(Map<String, Dataset> namedDatasets, Set<Component> ide
     }
 
     @Override
-    public Optional<Stream<DataPoint>> getData(Order requestedOrder, Dataset.Filtering filtering, Set<String> components) {
+    public Stream<DataPoint> computeData(Ordering requestedOrder, Filtering filtering, Set<String> components) {
 
         // Try to create a compatible order.
-        // If not, the caller will have to sort the result manually.
-        Optional<Order> compatibleOrder = createCompatibleOrder(getDataStructure(), getCommonIdentifiers(), requestedOrder);
-        if (!compatibleOrder.isPresent()) {
-            return Optional.empty();
-        }
-
-        Order requiredOrder = compatibleOrder.get();
+        Ordering requiredOrder = createCompatibleOrder(getDataStructure(), getCommonIdentifiers(), requestedOrder);
 
         // Compute the predicate
-        Order predicate = computePredicate(requiredOrder);
+        Ordering predicate = computePredicate(requiredOrder);
 
+        // TODO: Use abstract operation here.
         Iterator<Dataset> iterator = datasets.values().iterator();
         Dataset left = iterator.next();
         Dataset right = left;
 
-        // Close all children
+        ImmutableList.Builder<Stream<DataPoint>> originals = ImmutableList.builder();
+
         Closer closer = Closer.create();
         try {
 
             Table<Component, Dataset, Component> componentMapping = getComponentMapping();
```

Comment:
Can you use AbstractJoinOperation.getColumnMapping instead?

Reply:
Right. I am removing it.